### PR TITLE
[t2016] New drag and drop UI

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/b9a2a51f5b16c65/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/b9a2a51f5b16c65/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/7b04a849357d470/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/7b04a849357d470/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/135ce7c07bab1e4/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/135ce7c07bab1e4/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/44e989453ee50fe/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/44e989453ee50fe/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/4f4816c28dbb662/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/dc79617f49c9f59/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/4f4816c28dbb662/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/dc79617f49c9f59/output.txt

Total script time: 1.98 mins

Server is running on http://pr1089.butyr.org/templates/basic/
Run the Test Suite: http://pr1089.butyr.org/test/

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+ * jQuery UI
+ *
+ * Copyright 2012, AUTHORS.txt (http://jqueryui.com/about)
+ * Dual licensed under the MIT or GPL Version 2 licenses.
+ * http://jquery.org/license
+ *
+ * Modified by @k88hudson for Popcorn Maker
+ */
+
+ .transition(@transition1) {
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less]: Might as well stay consistent here and add spacing.

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  -o-transition: @transition1;
+  -ms-transition: @transition1;
+  transition: @transition1;
+}
+.box-sizing( @property ) {
+  // https://developer.mozilla.org/En/CSS/Box-sizing
+  -webkit-box-sizing: @property; // Chrome 10-
+  -moz-box-sizing: @property;
+  -ms-box-sizing: @property; // IE 8+
+  box-sizing: @property; // Opera, Safari 5.1+, Chrome 10+
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less:23]: Do we need all the comments?

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+
+@highlight: #3FB58E; //3FB58E;
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less:27]: Why the comment?

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+.ui-resizable-autohide 
+.ui-resizable-handle { 
+  display: none; 
+}
+
+
+.ui-resizable .ui-icon,
+[class^="ui-resizable-"] {
+  border: 3px solid @highlight;
+  background: darken(@highlight, 10% );
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less]: again, might as well stay consistent on the spacing

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+.ui-draggable-handle {
+  position: absolute;
+  z-index: 100;
+}
+.ui-draggable-handle {
+  cursor: move; 
+}
+.ui-draggable-top {
+  height: @offset * 2;
+  top: -@offset*2;
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less]: consistent spacing

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+  left: -@offset;
+  right: -@offset;
+}
+.ui-draggable-bottom {
+  height: @offset * 2;
+  bottom: -@offset*2;
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less]: ^

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+  left: -@offset;
+  right: -@offset;
+}
+.ui-draggable-left {
+  width: @offset * 2;
+  bottom: -@offset*2;
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less]: ^

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+  top: 0;
+  left: -@offset*2;
+}
+.ui-draggable-right {
+  width: @offset * 2;
+  bottom: -@offset*2;
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less]: ^

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+  height: @offset * 2;
+  bottom: -@offset*2;
+  left: -@offset;
+  right: -@offset;
+}
+.ui-draggable-left {
+  width: @offset * 2;
+  bottom: -@offset*2;
+  top: 0;
+  left: -@offset*2;
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less]: ^

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+}
+.ui-draggable-right {
+  width: @offset * 2;
+  bottom: -@offset*2;
+  top: -@offset*2;
+  right: -@offset*2;
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less]: And these 2

File: templates/assets/editors/editorhelper.js [hunk 11]
```diff
@@ -22,28 +22,34 @@
...
      */
     global.EditorHelper.draggable = function( trackEvent, dragContainer, mediaContainer, options ) {
-      var media = mediaContainer.getBoundingClientRect(),
-          iframeVideo = mediaContainer.querySelector( "iframe" );
+      var media = mediaContainer.getBoundingClientRect();
 
       options = options || {};
 
+      function createHelper( suffix ) {
+        var el = document.createElement( "div");
```
Comment (mjschranz) [inline templates/assets/editors/editorhelper.js]: Spacing

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+  height: 30px;
+  width: 30px;
+  border: none;
+  /*
+  border-radius: 2px 0 0 2px;
+  right: 100%;
+  margin-right: @offset + 1px;
+  top: 0;
+  margin-left: -15px;
+  */
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less]: Remove the commented block

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+  border-radius: 0;
+  top: -1px;
+  //margin-bottom: 1px;
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less]: remove comment

File: templates/assets/css/jquery-ui/jquery.ui.butter.less [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+    content: "";
+    display: inline-block;
+    height: 14px;
+    width: 14px;
+    margin-top: 7px;
+    margin-left: 8px;
+    background: url( "../../../../resources/glyphicons-halflings-green.png" ) -168px -72px;
+  }
+  &:hover {
+   // background-color: lighten( @highlight, 10% );
```
Comment (mjschranz) [inline templates/assets/css/jquery-ui/jquery.ui.butter.less]: Remove this hover style since it isn't doing anything

**Pull request merged**